tests/test_cli/test_generate_key.py

```diff
@@ -189,3 +191,48 @@ def teardown_class(cls):
         """Tear the test down."""
         os.chdir(cls.cwd)
         shutil.rmtree(cls.t)
+
+
+class TestGenerateKeyWithAddKeyWithoutConnection(AEATestCaseEmpty):
+    """Test that the command 'aea generate-key --add-key' works as expected."""
+
+    keys_config_path = "agent.private_key_paths"
+    args = []  # type: ignore
+
+    def test_fetchai(self):
+        """Test that the fetch private key is created correctly."""
+
+        with cd(self._get_cwd()):
+            result = self.run_cli_command(
+                "config", "get", self.keys_config_path, cwd=self._get_cwd()
+            )
+            assert result.exit_code == 0
+            assert json.loads(result.stdout_bytes) == {}
+
+            args = [*CLI_LOG_OPTION, "generate-key", FetchAICrypto.identifier]
+            result = self.run_cli_command(
+                *args, "--add-key", *self.args, cwd=self._get_cwd()
+            )
+            assert result.exit_code == 0
+            assert Path(FETCHAI_PRIVATE_KEY_FILE).exists()
+            make_crypto(
+                FetchAICrypto.identifier,
+                private_key_path=FETCHAI_PRIVATE_KEY_FILE,
+                password=None,
+            )
+
+            Path(FETCHAI_PRIVATE_KEY_FILE).unlink()
+
+            result = self.run_cli_command(
+                "config", "get", self.keys_config_path, cwd=self._get_cwd()
+            )
+            assert result.exit_code == 0
+            agent_keys = json.loads(result.stdout_bytes)
+            assert agent_keys.get(FetchAICrypto.identifier) == FETCHAI_PRIVATE_KEY_FILE
+
+
+class TestGenerateKeyWithAddKeyWithConnection(AEATestCaseEmpty):
```

Comment:
Is this complete?

Reply:
i think so

Reply:
There is no test in this class.

Reply:
Maybe you meant to subclass `TestGenerateKeyWithAddKeyWithoutConnection`?

Reply:
Yep

Reply:
sorry. missed this. added

Reply:
Still haven't :)